crates/polars-plan/src/plans/conversion/dsl_to_ir.rs

```diff
@@ -726,21 +709,59 @@ pub fn to_alp_impl(
     Ok(lp_arena.add(v))
 }
 
+/// Expand scan paths if they were not already expanded.
+#[allow(unused_variables)]
+fn expand_scan_paths(
+    paths: Arc<Mutex<(Arc<[PathBuf]>, bool)>>,
+    scan_type: &mut FileScan,
+    file_options: &mut FileScanOptions,
+) -> PolarsResult<Arc<[PathBuf]>> {
+    let mut lock = paths.lock().unwrap();
+
+    // Return if paths are already expanded
+    if lock.1 {
+        return Ok(lock.0.clone());
+    }
+
+    let paths_expanded = match scan_type {
+        #[cfg(feature = "parquet")]
+        FileScan::Parquet {
+            ref cloud_options, ..
+        } => expand_scan_paths_with_hive_update(&lock.0, file_options, cloud_options)?,
+        #[cfg(feature = "ipc")]
+        FileScan::Ipc {
+            ref cloud_options, ..
+        } => expand_scan_paths_with_hive_update(&lock.0, file_options, cloud_options)?,
+        #[cfg(feature = "csv")]
+        FileScan::Csv {
+            ref cloud_options, ..
+        } => expand_paths(&lock.0, file_options.glob, cloud_options.as_ref())?,
+        #[cfg(feature = "json")]
+        FileScan::NDJson { .. } => expand_paths(&lock.0, file_options.glob, None)?,
+        FileScan::Anonymous { .. } => lock.0.clone(), // Anonymous scans are already expanded.
```

Comment:
this branch should be `unreachable!`, given L722.

Reply:
Yes, but I couldn't get that to work nicely with feature flags. Since Anonymous is the only non-feature gated branch, it says the rest of the function is then unreachable / dead code. Best solution I could come up with is to just implement this branch.

Reply:
No, you could use the `unreachable!()` macro in that branch.

That way the future readers knows that it's a dead branch.

Reply:
I know you meant that - but it doesn't work due to feature flag shenanigans, as I tried to explain.

Feel free to push a commit with a different solution though.